shared/modules/treestore_module.go

```diff
@@ -0,0 +1,33 @@
+package modules
+
+import (
+	"github.com/jackc/pgx/v5"
+	"github.com/pokt-network/pocket/persistence/indexer"
+)
+
+const (
+	TreeStoreModuleName = "tree_store"
+)
+
+type TreeStoreOption func(TreeStoreModule)
+
+type TreeStoreFactory = FactoryWithOptions[TreeStoreModule, TreeStoreOption]
+
+// TreeStoreModules defines the interface for atomic updates and rollbacks to the internal
+// merkle trees that compose the state hash of pocket.
+
+type TreeStoreModule interface {
+	IntegratableModule
+
+	// Update returns the new state hash for a given height.
+	// * Height is passed through to the Update function and is used to query the TxIndexer for transactions
+	// to update into the merkle tree set
+	// * Passing a higher height will cause a change but repeatedly calling the same or a lower height will
+	// not incur a change.
+	// * By nature of it taking a pgx transaction at runtime, Update inherits the pgx transaction's read view of the
+	// database.
+	// TECHDEBT: Remove the TxIndexer pass here and get it from the bus
```

Comment:
Is this going to be done in a separate PR?

Also, the previous comment we have in the other file is clearer IMO:

```
// TODO(#808): Change interface to `Update(pgtx pgx.Tx, height uint64) (string, error)`
```

Can you make sure the other text/comments are properly mirrored as well.

Reply:
> @dylanlott Do you get value from these? 

No, good catch; I was able to remove that function because of the way @bryanchriswhite and I settled on to properly handle the TreeStore as a sub-module of Persistence.